### PR TITLE
Add the focus state related methods to the platform dispatcher

### PR DESCRIPTION
**Pull request by tugorez**
These changes were originally landed on https://github.com/flutter/engine/pull/49841 but reverted in https://github.com/flutter/engine/commit/0eb7b109f0f2988f31dbb5d5c14fd99b4316c407. I fixed the offending dart snippets and (think) they now will work without issues (I think I was able to verify them locally by manually patching my bin/cache/... copy of these files with these changes). 

Relevant Issues are:
* Design doc link: https://github.com/flutter/website/actions/runs/7560898849/job/20588395967
* Design doc: https://github.com/flutter/flutter/issues/141711
* Focus in web multiview: https://github.com/flutter/flutter/issues/137443

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by tugorez** [comment]
cc @zanderso

File: lib/ui/platform_dispatcher.dart [hunk 1]
```diff
@@ -314,15 +314,8 @@ class PlatformDispatcher {
   /// callback is invoked indicating the new view that has focus and the direction
   /// in which focus was received. For example, if focus is moved to the [FlutterView]
   /// with ID 2 in the forward direction (could be the result of pressing tab)
-  /// the callback receives the following [ViewFocusEvent]:
-  ///
-  /// ```dart
-  /// ViewFocusEvent(
```
Comment (goderbauer) [inline lib/ui/platform_dispatcher.dart]: For analyzing the snippets, dart:ui is imported as `import 'dart:ui' as ui`. So, you need to prefix everything coming from `dart:ui` with `ui.` in these samples.

https://github.com/flutter/flutter/blob/b209125d11affe911da7f9694aad78f87f1453d2/dev/bots/analyze_snippet_code.dart#L506

**Review by goderbauer — APPROVED** [review]
LGTM (would be nice to add the snippets back in a follow-up PR, though)

File: lib/ui/platform_dispatcher.dart (hunk 1 above)
Comment (tugorez) [reply lib/ui/platform_dispatcher.dart]: Interesting, these patterns are somewhat hidden. It is not intuitive what's expected.

Also, why don't we make these verifications as part of the engine submit process?

**Pull request merged**